src/daft-logical-plan/src/builder/resolve_expr.rs

```diff
@@ -151,6 +151,80 @@ fn col_resolves_to_plan(column: &UnresolvedColumn, plan: &LogicalPlanRef) -> Daf
     })
 }
 
+fn replace_element_with_column_ref(expr: ExprRef, replacement: ExprRef) -> DaftResult<ExprRef> {
+    expr.transform(|e| {
+        if matches!(
+            e.as_ref(),
+            Expr::Column(Column::Unresolved(UnresolvedColumn {
+                name,
+                plan_ref: PlanRef::Unqualified,
+                plan_schema: None
+            })) if name.as_ref() == ""
+        ) {
+            Ok(Transformed::yes(replacement.clone()))
+        } else {
+            Ok(Transformed::no(e))
+        }
+    })
+    .map(|res| res.data)
+}
+
+fn resolve_list_evals(expr: ExprRef) -> DaftResult<ExprRef> {
+    if expr.has_element() {
+        expr.transform_down(|e| {
+            let expr_ref = e.as_ref();
+            if let Expr::ScalarFunction(sf) = expr_ref {
+                // the `list` type should always be the first element
+                let inputs = sf.inputs.clone();
+                let list_col = inputs
+                    .first()
+                    .expect("list should have at least one element");
+
+                let exploded = list_col.clone().explode()?;
+
+                let mut new_inputs = Vec::with_capacity(inputs.len());
+                new_inputs.push(FunctionArg::Unnamed(list_col.clone()));
+
+                for input in inputs.iter().skip(1) {
+                    ensure!(
+                       !matches!(input.inner().as_ref(), Expr::Column(..)),
+                       ValueError: "unexpected column reference"
+
+                    );
+
+                    let replaced = match input {
+                        daft_dsl::functions::FunctionArg::Named { name, arg } => {
+                            daft_dsl::functions::FunctionArg::Named {
+                                name: name.clone(),
+                                arg: replace_element_with_column_ref(
+                                    arg.clone(),
+                                    exploded.clone(),
+                                )?,
+                            }
+                        }
+                        daft_dsl::functions::FunctionArg::Unnamed(arg) => {
+                            daft_dsl::functions::FunctionArg::Unnamed(
+                                replace_element_with_column_ref(arg.clone(), exploded.clone())?,
+                            )
+                        }
+                    };
+                    new_inputs.push(replaced);
+                }
+                let sf = ScalarFunction {
+                    udf: sf.udf.clone(),
+                    inputs: FunctionArgs::new_unchecked(new_inputs),
+                };
+                Ok(Transformed::yes(Expr::ScalarFunction(sf).arced()))
```

Comment:
We can skip all of the children
```suggestion
                Ok(Transformed::new(Expr::ScalarFunction(sf).arced(), true, TreeNodeRecursion::Jump))
```

Reply:
what would happen if you had multiple list map functions? wouldnt this skip handling the children?

Reply:
Actually yeah. I was thinking this would not happen since we do not allow other column expressions in here, but we could conceivably have a List[List[T]] type expression, where you would do a nested list map